shared/src/ContentReactor.Shared/BlobRepository/BlobRepository.cs

```diff
@@ -1,132 +1,139 @@
+            await container.CreateIfNotExistsAsync();
 
-            // list all blobs in folder
-            var blobsInFolder = new List<CloudBlockBlob>();
-            var continuationToken = new BlobContinuationToken();
-            do
+            var blobs = new List<BlobItem>();
+            await foreach (var blob in container.GetBlobsAsync())
             {
-                var currentPage = await folder.ListBlobsSegmentedAsync(true, BlobListingDetails.Metadata, 100, continuationToken, null, null);
-                blobsInFolder.AddRange(currentPage.Results.OfType<CloudBlockBlob>());
-                continuationToken = currentPage.ContinuationToken;
-            } while (continuationToken != null);
+                blobs.Add(blob);
+            }
 
-            return blobsInFolder;
+            return blobs;
         }
 
-        public Task<bool> BlobExistsAsync(CloudBlockBlob blob)
+        public Task<Response<bool>> BlobExistsAsync(BlockBlobClient blob)
         {
             return blob.ExistsAsync();
         }
 
-        public Task DeleteBlobAsync(string containerName, string folderName, string blobId)
+        public async Task DeleteBlobAsync(string containerName, string blobName)
         {
-            var container = BlobClient.GetContainerReference(containerName);
-            var folder = container.GetDirectoryReference(folderName);
-            var blob = folder.GetBlockBlobReference(blobId);
-            return blob.DeleteIfExistsAsync();
+            var container = BlobServiceClient.GetBlobContainerClient(containerName);
+            await container.CreateIfNotExistsAsync();
+            var blob = container.GetBlockBlobClient(blobName);
+
+            await blob.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots);
         }
 
-        public async Task UpdateBlobMetadataAsync(CloudBlockBlob blob)
+        public async Task UpdateBlobMetadataAsync(BlockBlobClient blob, IDictionary<string, string> metaData)
         {
-            var metadataUpdate = blob.SetMetadataAsync();
-            var propertiesUpdate = blob.SetPropertiesAsync();
-            await Task.WhenAll(metadataUpdate, propertiesUpdate);
+            await blob.SetMetadataAsync(metaData);
         }
 
-        public string GetSasTokenForBlob(CloudBlockBlob blob, SharedAccessBlobPolicy sasPolicy)
+        public string GetSasTokenForBlob(BlockBlobClient blob, BlobSasBuilder sasBuilder)
         {
-            var sasBlobToken = blob.GetSharedAccessSignature(sasPolicy);
-            return blob.Uri + sasBlobToken;
+            // Create a SharedKeyCredential that we can use to sign the SAS token
+            var credential = new StorageSharedKeyCredential(StorageAccountName, StorageAccountKey);
+
+            // Build a SAS URI
+            UriBuilder sasUri = new UriBuilder(blob.Uri)
+            {
+                Query = sasBuilder.ToSasQueryParameters(credential).ToString()
+            };
+
+            return sasUri.Uri.ToString();
```

Comment:
Always prefer `Uri.AbsoluteUri` over `Uri.ToString()` in .NET.  You'll run into weird encoding issues otherwise.

Reply:
Resolved